python/lsst/sims/GalSimInterface/galSimInterpreter.py

```diff
@@ -509,6 +520,67 @@ def writeImages(self, nameRoot=None):
 
         return namesWritten
 
+    def open_centroid_file(self, detector_name):
+        """
+        Open a centroid file.  This file will have one line per-object and the
+        it will be labeled with the objectID and then followed by the average X
+        Y position of the photons from the object. Either the true photon
+        position or the average of the pixelated electrons collected on a finite
+        sensor can be chosen.
+        """
+
+        visitID = self.obs_metadata.OpsimMetaData['obshistID']
+        file_name = self.centroid_base_name + str(visitID) + '_' + detector_name + '.txt'
+
+        self.centroid_handles[detector_name] = open(file_name, 'w')
+        self.centroid_handles[detector_name].write('{:15} {:>15} {:>10} {:>10}\n'.
+                                                   format('SourceID', 'Flux', 'xPix', 'yPix'))
+
+    def _writeObjectToCentroidFile(self, detector_name, bandpass_name, uniqueId, flux, xPix, yPix):
+        """
+        Write the flux and the the object position on the sensor for this object
+        into a centroid file.  First check if a centroid file exists for this
+        detector and, if it doesn't create it.
+
+        @param [in] detectorName is the name of the sensor the gsObject falls on.
+
+        @param [in] bandpassName is the name of the filter used in this exposure.
+
+        @param [in] UniqueID is the Unique ID of the gsObject.
+
+        @param [in] flux is the calculated flux for the gsObject in the given bandPass.
+        """
+
+        centroid_name = detector_name + '_' + bandpass_name
+
+        # If we haven't seen this sensor before open a centroid file for it.
+        if centroid_name not in self.centroid_handles:
+            self.open_centroid_file(centroid_name)
+
+        # Write the object to the file
+        self.centroid_handles[centroid_name].write('{:<15d} {:15.5f} {:10.2f} {:10.2f}\n'.
+                                                   format(uniqueId, flux, xPix, yPix))
+
+    def write_centroid_files(self):
+        """
+        Write the centroid data structure out to the files.
+
+        This funnction loops over the entries in the centroid list and
+        then sends them each to be writen to a file. The
+        _writeObjectToCentroidFile will decide how to put them in files.
+        """
+        # Loop over entries
+        for centroid_tuple in self.centroid_list:
+            (detector_name, bandpass_name, uniqueId, flux, xPix, yPix) = centroid_tuple
```

Comment:
I would skip setting these temporary variables and just do
```
           self._writeObjectToCentroidFile(*centroid_tuple)
```
for the line below.